catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPlugin.java

```diff
@@ -39,92 +45,89 @@
-            List<Serializable> validationWarnings = new LinkedList<>();
-            List<Serializable> validationErrors = new LinkedList<>();
-            // Default to allowing metacard through
-            Boolean toReturn = true;
-            // Run metacard through each validator
-            for (MetacardValidator metacardValidator : metacardValidators) {
-                try {
-                    // Attempt validation
-                    metacardValidator.validate(metacard);
-                } catch (ValidationException e) {
-                    // If validator is not explicitly turned on by admin, set invalid and allow through
-                    if (checkEnforcedMetacardValidators(metacardValidator)) {
-                        boolean validationErrorsExist = e.getErrors() != null && !e.getErrors()
-                                .isEmpty();
-                        boolean validationWarningsExist =
-                                e.getWarnings() != null && !e.getWarnings()
-                                        .isEmpty();
-                        if (validationErrorsExist || validationWarningsExist) {
-                            // Check for warnings and errors
-                            if (validationErrorsExist) {
-                                validationErrors.addAll(e.getErrors());
-                            }
-                            if (validationWarningsExist) {
-                                validationWarnings.addAll(e.getWarnings());
-                            }
-                        } else {
-                            LOGGER.error(
-                                    "Metacard validator {} did not have any warnings or errors but it threw a validation exception."
-                                            + " There is likely something wrong with your implementation. This will result in the metacard not"
-                                            + " being properly marked as invalid.",
-                                    getValidatorName(metacardValidator));
-                        }
-
-                    } else {
-                        // If validator is explicitly turned on, break out and do not include in the
-                        // list of metacards that will be allowed through.
-                        toReturn = false;
-                        break;
-                    }
+    @Override
+    public UpdateRequest process(UpdateRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        List<Map.Entry<Serializable, Metacard>> validatedMetacards =
+                getValidatedMetacards(input.getUpdates(), Map.Entry::getValue);
+        return new UpdateRequestImpl(validatedMetacards,
+                input.getAttributeName(),
+                input.getProperties(),
+                input.getStoreIds());
+    }
 
+    private <T> List<T> getValidatedMetacards(List<T> requestItems,
+            Function<T, Metacard> itemToMetacard) {
```

Comment:
could you explain a little bit what all this fancy java 8 stuff is doing? like what is itemToMetacard?